// @chipspc/vant-dgg/es/evaluate/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import { createNamespace, isDef } from '../utils';
import Field from '../field';
import Button from '../button';
import Row from '../row';
import Col from '../col';

var _createNamespace = createNamespace('evaluate'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

export default createComponent({
  props: {
    //按钮文字
    buttonText: String,
    //选中值
    value: String,
    //是否超出隐藏
    ellipsis: Boolean,
    // 输入框绑定值
    inputValue: String,
    // 占位符
    placeholder: String,
    // 是否loading
    isSaving: Boolean,
    // 传入的数组
    options: {
      type: Array,
      default: function _default() {
        return [];
      }
    }
  },
  data: function data() {
    return {
      data: '',
      categoryIndex: 0,
      categoryId: this.options[0].id || null
    };
  },
  watch: {
    inputValue: {
      handler: function handler(val) {
        this.data = val;
      },
      immediate: true
    }
  },
  methods: {
    onSave: function onSave() {
      this.$emit('save', _extends({
        value: this.data,
        id: this.categoryId
      }));
    },
    onChoose: function onChoose(id, index) {
      this.categoryIndex = index;
      this.categoryId = id;
      this.$emit('choose', id);
    },
    onInput: function onInput(val) {
      this.data = val;
    }
  },
  beforeCreate: function beforeCreate() {},
  render: function render() {
    var _this = this;

    var h = arguments[0];
    var data = this.data;
    var Options = this.options.map(function (option, index) {
      return h(Col, {
        "attrs": {
          "span": "12"
        },
        "key": option.id || null,
        "on": {
          "click": _this.onChoose.bind(_this, option.id, index)
        }
      }, [h("div", {
        "class": [bem('text'), _this.categoryIndex == index && bem('active')]
      }, [h("span", {
        "class": _this.ellipsis && bem('ellipsis')
      }, [option.text])])]);
    });
    return h("div", [h(Row, {
      "attrs": {
        "gutter": "12"
      },
      "class": bem('list')
    }, [Options]), h(Field, {
      "attrs": {
        "border": true,
        "clearable": true,
        "value": data,
        "placeholder": this.placeholder
      },
      "on": {
        "input": this.onInput
      },
      "class": bem('field')
    }), h("div", {
      "class": bem('buttons')
    }, [h(Button, {
      "attrs": {
        "block": true,
        "loading": this.isSaving,
        "type": "primary",
        "text": this.buttonText || '确定'
      },
      "on": {
        "click": this.onSave
      }
    })])]);
  }
});
