// @chipspc/vant-dgg/lib/work-tab-filter/index.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _utils = require("../utils");

var _relation = require("../mixins/relation");

var _createNamespace = (0, _utils.createNamespace)('work-tab-filter'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

var _default = createComponent({
  mixins: [(0, _relation.ParentMixin)('SpWorkTabFilter')],
  props: {
    value: {
      type: [Number, String],
      default: 0
    }
  },
  data: function data() {
    return {
      itemIndex: Number(this.value)
    };
  },
  computed: {},
  watch: {
    itemIndex: function itemIndex(_itemIndex) {
      this.setActiveItem(_itemIndex);
    }
  },
  mounted: function mounted() {
    this.setActiveItem(this.itemIndex);
  },
  methods: {
    setActiveItem: function setActiveItem(itemIndex) {
      this.children.forEach(function (item, index) {
        item.active = (item.name || index) === itemIndex;
      });
    },
    onChange: function onChange(active, index) {
      if (active !== this.itemIndex) {
        this.$emit('change', active, index);
      }
    }
  },
  render: function render() {
    var h = arguments[0];
    return h("div", {
      "class": bem()
    }, [this.slots()]);
  }
});

exports.default = _default;
