// @chipspc/vant-dgg/lib/bottombar-button/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _utils = require("../utils");

var _router = require("../utils/router");

var _relation = require("../mixins/relation");

var _button = _interopRequireDefault(require("../button"));

/*
 * @Author: zwh
 * @Date: 2020-10-09 16:35:48
 * @LastEditTime: 2020-10-15 10:43:17
 * @FilePath: \vant\src\bottombar-button\index.js
 */
var _createNamespace = (0, _utils.createNamespace)('bottombar-button'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

var _default = createComponent({
  mixins: [(0, _relation.ChildrenMixin)('SpBottombar')],
  props: (0, _extends2.default)({}, _router.routeProps, {
    //路由跳转属性 url to replace
    type: String,
    //按钮类型
    text: String,
    //按钮文本
    icon: String,
    //左侧图标名称或图片链接
    color: String,
    //按钮颜色，支持传入 linear-gradient 渐变色
    loading: Boolean,
    //是否显示为加载状态
    disabled: Boolean,
    //是否禁用按钮
    buttonTextSlot: Object //多层按钮文本

  }),
  computed: {},
  methods: {
    onClick: function onClick(event) {
      //抛出点击事件
      this.$emit('click', event); //路由跳转

      (0, _router.route)(this.$router, this);
    },
    //生成多层按钮文本node
    buttonSlot: function buttonSlot() {
      var h = this.$createElement;
      var slot = this.buttonTextSlot;

      if (!!slot) {
        var vnode = h("slot", [h("div", {
          "class": bem('slot-top')
        }, [slot.top]), h("div", {
          "class": bem('slot-bottom')
        }, [slot.bottom])]);
        return vnode;
      } else {
        return this.slots() || this.text;
      }
    }
  },
  render: function render() {
    var h = arguments[0];
    return h(_button.default, {
      "class": bem([{
        first: true
      }, this.type]),
      "attrs": {
        "size": "large",
        "type": this.type,
        "icon": this.icon,
        "color": this.color,
        "loading": this.loading,
        "disabled": this.disabled
      },
      "on": {
        "click": this.onClick
      }
    }, [this.buttonSlot()]);
  }
});

exports.default = _default;
