// @chipspc/vant-dgg/lib/evaluate/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _utils = require("../utils");

var _field = _interopRequireDefault(require("../field"));

var _button = _interopRequireDefault(require("../button"));

var _row = _interopRequireDefault(require("../row"));

var _col = _interopRequireDefault(require("../col"));

var _createNamespace = (0, _utils.createNamespace)('evaluate'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

var _default2 = createComponent({
  props: {
    //按钮文字
    buttonText: String,
    //选中值
    value: String,
    //是否超出隐藏
    ellipsis: Boolean,
    // 输入框绑定值
    inputValue: String,
    // 占位符
    placeholder: String,
    // 是否loading
    isSaving: Boolean,
    // 传入的数组
    options: {
      type: Array,
      default: function _default() {
        return [];
      }
    }
  },
  data: function data() {
    return {
      data: '',
      categoryIndex: 0,
      categoryId: this.options[0].id || null
    };
  },
  watch: {
    inputValue: {
      handler: function handler(val) {
        this.data = val;
      },
      immediate: true
    }
  },
  methods: {
    onSave: function onSave() {
      this.$emit('save', (0, _extends2.default)({
        value: this.data,
        id: this.categoryId
      }));
    },
    onChoose: function onChoose(id, index) {
      this.categoryIndex = index;
      this.categoryId = id;
      this.$emit('choose', id);
    },
    onInput: function onInput(val) {
      this.data = val;
    }
  },
  beforeCreate: function beforeCreate() {},
  render: function render() {
    var _this = this;

    var h = arguments[0];
    var data = this.data;
    var Options = this.options.map(function (option, index) {
      return h(_col.default, {
        "attrs": {
          "span": "12"
        },
        "key": option.id || null,
        "on": {
          "click": _this.onChoose.bind(_this, option.id, index)
        }
      }, [h("div", {
        "class": [bem('text'), _this.categoryIndex == index && bem('active')]
      }, [h("span", {
        "class": _this.ellipsis && bem('ellipsis')
      }, [option.text])])]);
    });
    return h("div", [h(_row.default, {
      "attrs": {
        "gutter": "12"
      },
      "class": bem('list')
    }, [Options]), h(_field.default, {
      "attrs": {
        "border": true,
        "clearable": true,
        "value": data,
        "placeholder": this.placeholder
      },
      "on": {
        "input": this.onInput
      },
      "class": bem('field')
    }), h("div", {
      "class": bem('buttons')
    }, [h(_button.default, {
      "attrs": {
        "block": true,
        "loading": this.isSaving,
        "type": "primary",
        "text": this.buttonText || '确定'
      },
      "on": {
        "click": this.onSave
      }
    })])]);
  }
});

exports.default = _default2;
