// @chipspc/vant-dgg/es/work-tab/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import { createNamespace } from '../utils';
import { ChildrenMixin } from '../mixins/relation';
import { routeProps } from '../utils/router';

var _createNamespace = createNamespace('work-tab'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

export default createComponent({
  mixins: [ChildrenMixin('SpWorkTabs')],
  props: _extends({}, routeProps, {
    //标注点
    dot: Boolean,
    //标签名
    name: [Number, String],
    //标签信息
    info: [Number, String],
    //标注数字
    badge: [Number, String],
    title: String,
    //标签样式
    titleStyle: null,
    //不可选中的boolean
    disabled: Boolean,
    //是否需要内容
    needContent: {
      type: Boolean,
      default: true
    }
  }),
  data: function data() {
    return {
      inited: false
    };
  },
  computed: {
    //当前选中的index
    computedName: function computedName() {
      var _this$name;

      return (_this$name = this.name) != null ? _this$name : this.index;
    },

    /**
     *
     * 计算是否是被选中
     */
    isActive: function isActive() {
      var active = this.computedName === this.parent.currentName;

      if (active) {
        this.inited = true;
      }

      return active;
    }
  },
  watch: {
    //监听当前的title
    title: function title() {
      this.parent.setLine();
    },

    /**
     *初始化标签
     *
     * @param {*} val
     */
    inited: function inited(val) {
      var _this = this;

      if (this.parent.lazyRender && val) {
        this.$nextTick(function () {
          _this.parent.$emit('rendered', _this.computedName, _this.title);
        });
      }
    }
  },
  render: function render(h) {
    var slots = this.slots,
        parent = this.parent,
        isActive = this.isActive;
    var shouldRender = this.inited || parent.scrollspy || !parent.lazyRender;
    var show = parent.scrollspy || isActive;
    var Content = shouldRender ? slots() : h();

    if (parent.animated) {
      return this.needContent && h("div", {
        "attrs": {
          "role": "tabpanel",
          "aria-hidden": !isActive
        },
        "class": bem('pane-wrapper', {
          inactive: !isActive
        })
      }, [h("div", {
        "class": bem('pane')
      }, [Content])]);
    }

    return this.needContent && h("div", {
      "directives": [{
        name: "show",
        value: show
      }],
      "attrs": {
        "role": "tabpanel"
      },
      "class": bem('pane')
    }, [Content]);
  }
});
