// @chipspc/vant-dgg/lib/work-tab-filter-item/index.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _utils = require("../utils");

var _relation = require("../mixins/relation");

// Utils
// Mixins
var _createNamespace = (0, _utils.createNamespace)('work-tab-filter-item'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

var _default = createComponent({
  mixins: [(0, _relation.ChildrenMixin)('SpWorkTabFilter')],
  props: {},
  data: function data() {
    return {
      active: false
    };
  },
  computed: {},
  methods: {
    handleItemClick: function handleItemClick() {
      this.parent.onChange(this.slots()[0].text, this.name || this.index);
      this.parent.itemIndex = this.index;
    }
  },
  render: function render() {
    var h = arguments[0];
    return h("div", {
      "class": bem({
        active: this.active
      }),
      "on": {
        "click": this.handleItemClick
      }
    }, [h("div", {
      "class": bem('item')
    }, [this.slots(), h("span", {
      "class": bem('span')
    })])]);
  }
});

exports.default = _default;
