// @chipspc/vant-dgg/es/bottombar-info/index.js
/*
 * @Author: your name
 * @Date: 2020-10-09 16:35:48
 * @LastEditTime: 2020-10-15 16:12:10
 * @LastEditors: Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: \vant\src\bottombar-info\index.js
 */
import { createNamespace, addUnit } from '../utils';
import { ChildrenMixin } from '../mixins/relation';
import Icon from '../icon';

var _createNamespace = createNamespace('bottombar-info'),
    createComponent = _createNamespace[0],
    bem = _createNamespace[1];

export default createComponent({
  mixins: [ChildrenMixin('SpBottombar')],
  props: {
    card: {
      type: Object
    },
    fit: {
      type: String,
      default: 'cover'
    }
  },
  computed: {},
  methods: {
    Avatar: function Avatar() {
      var h = this.$createElement;
      var card = this.card || {};

      if (card.imgSrc) {
        var size = addUnit(card.avatarSize);
        var imgNode = card.imgSrc ? h("img", {
          "style": {
            width: size,
            height: size,
            borderRadius: card.round ? '999px' : false,
            objectFit: this.fit
          },
          "attrs": {
            "src": card.imgSrc
          }
        }) : '';
        return h("div", {
          "class": bem('avatar', {
            round: card.round
          }),
          "style": {
            width: size,
            height: size
          }
        }, [imgNode]);
      }
    },
    Title: function Title() {
      var h = this.$createElement;
      var card = this.card || {};
      var name = card.cardName ? h("div", {
        "class": bem('card')
      }, [h("span", {
        "class": bem('name')
      }, [card.cardName || '']), h(Icon, {
        "attrs": {
          "tag": "span",
          "size": card.size,
          "name": card.icon || ''
        }
      })]) : '';
      var sign = card.cardSign ? h("p", {
        "class": bem('sign')
      }, [card.cardSign || '']) : '';
      return h("div", {
        "class": name && sign ? bem('content') : ''
      }, [name, sign]);
    }
  },
  render: function render() {
    var h = arguments[0];
    if (!this.card) return '';
    return h("div", {
      "class": bem()
    }, [this.Avatar(), this.Title()]);
  }
});
